statusHandler/persistor/persistentHandler.go

```diff
@@ -0,0 +1,209 @@
+package persistor
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go/core"
+	"github.com/ElrondNetwork/elrond-go/core/logger"
+	"github.com/ElrondNetwork/elrond-go/marshal"
+	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/storage"
+	"github.com/ElrondNetwork/elrond-go/storage/storageUnit"
+)
+
+const saveInterval = 10 * time.Second
+
+//StatusMetricsDbEntry is key used to save metrics in storage
+const StatusMetricsDbEntry = "status_metrics_db_entry"
+
+var log = logger.DefaultLogger()
+
+// PersistentStatusHandler is a status handler that will save metrics in storage
+type PersistentStatusHandler struct {
+	store             storage.Storer
+	persistentMetrics *sync.Map
+	marshalizer       marshal.Marshalizer
+}
+
+// NewPersistentStatusHandler will return an instance of the persistent status handler
+func NewPersistentStatusHandler(
+	marshalizer marshal.Marshalizer,
+) (*PersistentStatusHandler, error) {
+	if marshalizer == nil || marshalizer.IsInterfaceNil() {
+		return nil, process.ErrNilMarshalizer
+	}
+
+	psh := new(PersistentStatusHandler)
+	psh.store = storageUnit.NewNilStorer()
+	psh.marshalizer = marshalizer
+	psh.persistentMetrics = &sync.Map{}
+	psh.initMap()
+	return psh, nil
+}
+
+func (psh *PersistentStatusHandler) initMap() {
+	initUint := uint64(0)
+
+	psh.persistentMetrics.Store(core.MetricCountConsensus, initUint)
+	psh.persistentMetrics.Store(core.MetricCountConsensusAcceptedBlocks, initUint)
+	psh.persistentMetrics.Store(core.MetricCountAcceptedBlocks, initUint)
+	psh.persistentMetrics.Store(core.MetricCountLeader, initUint)
+	psh.persistentMetrics.Store(core.MetricNumProcessedTxs, initUint)
+	psh.persistentMetrics.Store(core.MetricNumShardHeadersProcessed, initUint)
+}
+
+// StartStoreMetricsInStorage will start save status metrics in storage at a constant interval
+func (psh *PersistentStatusHandler) StartStoreMetricsInStorage() {
+	go func() {
+		time.Sleep(saveInterval)
```

Comment:
Why it is necessary to sleep twice or the first save?

Reply:
you right. It is enough to sleep just one time.